backends/tfhe-cuda-backend/cuda/src/crypto/keyswitch.cuh

```diff
@@ -76,15 +101,16 @@ __host__ void cuda_keyswitch_lwe_ciphertext_vector(
     uint32_t base_log, uint32_t level_count, uint32_t num_samples) {
 
   cudaSetDevice(gpu_index);
-  constexpr int ideal_threads = 1024;
-  if (lwe_dimension_out + 1 > ideal_threads)
-    PANIC("Cuda error (keyswitch): lwe dimension size out should be greater "
-          "or equal to the number of threads per block")
+
+  constexpr int num_threads_x = 16;
+  constexpr int num_threads_y = 32;
```

Comment:
Do we really need to hard code the block dimension?

Reply:
If so, why so small values for each dimension?

Reply:
You have to consider that it is a 2D block, so the total number of threads in the block is 16x32 = 512. 
The number of threads in the y direction is hardcoded since it is the one that follows the reduction logic.

Reply:
I think it should work as it is indeed. I remember seeing parameter sets with N=256 but don't think they are in use nowadays.